src/plugins/FileService/components/Uploading.tsx

```diff
@@ -0,0 +1,100 @@
+import { useAsync, useBeforeUnload } from 'react-use'
+import { Image } from '../../../components/shared/Image'
+import Services, { ServicesWithProgress } from '../../../extension/service'
+import { getUrl } from '../../../utils/utils'
+import { pluginId } from '../constants'
+import { useExchange } from '../hooks/Exchange'
+import type { FileInfo } from '../types'
+import { FileName } from './FileName'
+import { ProgressBar } from './ProgressBar'
+
+const useStyles = makeStyles({
+    container: {
+        height: 250,
+        flexDirection: 'column',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        userSelect: 'none',
+        paddingTop: 18,
+        paddingBottom: 18,
+    },
+    name: {
+        fontSize: 16,
+        lineHeight: 1.75,
+        textAlign: 'center',
+        color: '#3B3B3B',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+        width: 400,
+    },
+})
+
+interface RouteState {
+    key: string | undefined
+    name: string
+    size: number
+    type: string
+    block: Uint8Array
+    checksum: string
+}
+
+export const Uploading: React.FC = () => {
+    const classes = useStyles()
+    const history = useHistory()
+    const { onUploading } = useExchange()
+    const [startedAt] = React.useState(Date.now())
+    const [preparing, setPreparing] = React.useState(true)
+    const [sendSize, setSendSize] = React.useState(0)
+    const { state } = useLocation<RouteState>()
+    useBeforeUnload(true, 'uploading')
+    React.useEffect(() => {
+        onUploading(true)
+        return () => onUploading(false)
+    }, [onUploading])
+    useAsync(async () => {
+        const payloadTxID = await Services.Plugin.invokePlugin(pluginId, 'makeAttachment', {
+            key: state.key,
+            block: state.block,
+            type: state.type,
+        })
+        setPreparing(false)
+        for await (const pctComplete of ServicesWithProgress.pluginArweaveUpload(payloadTxID)) {
+            setSendSize(state.size * (pctComplete / 100))
+        }
+        const landingTxID = await Services.Plugin.invokePlugin(pluginId, 'uploadLandingPage', {
+            name: state.name,
+            size: state.size,
+            txId: payloadTxID,
+            type: state.type,
+            key: state.key,
+        })
+        const item: FileInfo = {
+            type: 'arweave',
+            id: state.checksum,
+
+            name: state.name,
+            size: state.size,
+            createdAt: new Date(startedAt),
+            key: state.key,
+            payloadTxID: payloadTxID,
+            landingTxID: landingTxID,
+        }
+        await Services.Plugin.invokePlugin(pluginId, 'setFileInfo', item)
+        history.push('/uploaded', item)
+    }, [])
+    return (
+        <Grid container className={classes.container}>
+            <Grid item>
+                <Image src={getUrl('/plugin/file-service/file.svg')} width={96} height={120} />
```

Comment:
Please don't use this utility to render SVG. You can change the SVG into a TSX file and get a high-resolution icon. And if possible, please use the icon from [@material-ui/icons](https://material-ui.com/components/material-icons/) or [react-feather](https://feathericons.com/).